snap/snapcraft.yaml

```diff
@@ -90,60 +92,85 @@ parts:
+      - enable-patchelf
     override-build: |
-      snapcraftctl build
+      ${SNAP}/libexec/snapcraft/craftctl default
 
       echo "Create libsodium symlink..."
       TRIPLET_PATH="$SNAPCRAFT_PART_INSTALL/usr/lib/$(gcc -print-multiarch)"
       LIBSODIUM="$(readlink -n "$TRIPLET_PATH/libsodium.so.23")"
       # Remove so the link can be recreated on re-builds
       rm -f "$TRIPLET_PATH/libsodium.so"
       ln -s "$LIBSODIUM" "$TRIPLET_PATH/libsodium.so"
-      $SNAPCRAFT_PROJECT_DIR/tools/snapcraft-override-build.sh
+
+      # Restore patched files
+      PYTHON_PACKAGE_PATH="${SNAPCRAFT_PART_INSTALL}/usr/lib/python3.10/"
+      CTYPES_INIT="ctypes/__init__.py"
+      CTYPES_INIT_ORIG="patched/${CTYPES_INIT}.orig"
+      [ -f "${CTYPES_INIT_ORIG}" ] && mv "${CTYPES_INIT_ORIG}" "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}"
+
+      # Apply patches
+      echo "Patching ctypes..."
+      patch -s -b "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}" "${SNAPCRAFT_PROJECT_DIR}/patches/ctypes_init.diff"
+
+      # Save patches to allow rebuilding
+      mkdir -p patched/ctypes
+      if [ -f "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}.orig" ]; then
+        mv "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}.orig" patched/ctypes
+      fi
+
+      # Disable site packages
+      sed -i "${SNAPCRAFT_PART_INSTALL}/usr/lib/python3.10/site.py" \
+        -e 's/^ENABLE_USER_SITE = None$/ENABLE_USER_SITE = False/'
 
   snapcraft:
     source: .
     plugin: python
     python-packages:
         - wheel
         - pip
-        - setuptools<66  # Limited to < 66 because we need `pkg_resources` and because `python-apt` doesn't build with the latest.
-    requirements:
+        # Limited to < 66 because we need `pkg_resources` and because `python-apt`
+        # does not build with the latest.
+        - setuptools<66
+    python-requirements:
         - requirements.txt
     organize:
-        # Put snapcraftctl into its own directory that can be included in the PATH
+        # Put snapcraftctl and craftctl into its own directory that can be included in the PATH
         # without including other binaries.
+        bin/craftctl: libexec/snapcraft/craftctl
         bin/snapcraftctl: bin/scriptlet-bin/snapcraftctl
         # Also install the compatibility wrapper for core22+
         bin/snapcraftctl-compat: libexec/snapcraft/snapcraftctl
     build-environment:
-        - "PIP_NO_BINARY": "PyNaCl"  # Build PyNaCl from source since the wheel files interact strangely with classic snaps.
-        - "SODIUM_INSTALL": "system"  # Use base image's libsodium for PyNaCl
-        - "CFLAGS": "$(pkg-config python-3.8 yaml-0.1 --cflags)"
+        # Build PyNaCl from source since the wheel files interact strangely with classic snaps.
+        - "PIP_NO_BINARY": "PyNaCl"
+        # Use base image's libsodium for PyNaCl
+        - "SODIUM_INSTALL": "system"
+        - "CFLAGS": "$(pkg-config python-3.10 yaml-0.1 --cflags)"
     override-build: |
-        snapcraftctl build
+        set -x
```

Comment:
Scriplet execution should be doing this automatically.

Reply:
yeah, this was to debug something else (the pre scriptlet execution itself)